Index: src/frontends/q4banking/lib/qbanking.h
===================================================================
--- src/frontends/q4banking/lib/qbanking.h	(revision 1972)
+++ src/frontends/q4banking/lib/qbanking.h	(working copy)
@@ -35,12 +35,12 @@
 #include <aqbanking/banking.h>
 #include <aqbanking/accstatus.h>
 
-#include <Qt/qobject.h>
-#include <Qt/qdatetime.h>
-#include <Qt/qstring.h>
-#include <Qt/qstringlist.h>
-#include <Qt/qwidget.h> // needed for QPointer<QWidget>
-#include <Qt/qpointer.h>
+#include <QtCore/qobject.h>
+#include <QtCore/qdatetime.h>
+#include <QtCore/qstring.h>
+#include <QtCore/qstringlist.h>
+#include <QtGui/qwidget.h> // needed for QPointer<QWidget>
+#include <QtCore/qpointer.h>
 
 #include <list>
 
Index: src/frontends/q4banking/lib/dialogs/qbimporter.h
===================================================================
--- src/frontends/q4banking/lib/dialogs/qbimporter.h	(revision 1972)
+++ src/frontends/q4banking/lib/dialogs/qbimporter.h	(working copy)
@@ -24,7 +24,7 @@
 #include <list>
 #include <string>
 
-#include <Qt/qstring.h>
+#include <QtCore/qstring.h>
 
 #include <gwenhywfar/logger.h>
 
Index: src/frontends/q4banking/lib/widgets/qbplugindescrlist.cpp
===================================================================
--- src/frontends/q4banking/lib/widgets/qbplugindescrlist.cpp	(revision 1972)
+++ src/frontends/q4banking/lib/widgets/qbplugindescrlist.cpp	(working copy)
@@ -17,7 +17,7 @@
 
 #include "qbplugindescrlist.h"
 #include <assert.h>
-#include <Qt/qstring.h>
+#include <QtCore/qstring.h>
 
 
 
Index: src/frontends/q4banking/lib/widgets/qbaccountlist.cpp
===================================================================
--- src/frontends/q4banking/lib/widgets/qbaccountlist.cpp	(revision 1972)
+++ src/frontends/q4banking/lib/widgets/qbaccountlist.cpp	(working copy)
@@ -17,7 +17,7 @@
 
 #include "qbaccountlist.h"
 #include <assert.h>
-#include <Qt/qstring.h>
+#include <QtCore/qstring.h>
 
 
 
Index: src/frontends/q4banking/lib/widgets/qbcfgtabpage.h
===================================================================
--- src/frontends/q4banking/lib/widgets/qbcfgtabpage.h	(revision 1972)
+++ src/frontends/q4banking/lib/widgets/qbcfgtabpage.h	(working copy)
@@ -18,8 +18,8 @@
 #include <aqbanking/banking.h>
 #include <q4banking/qbanking.h>
 
-#include <Qt/qwidget.h>
-#include <Qt/qstring.h>
+#include <QtGui/qwidget.h>
+#include <QtCore/qstring.h>
 //Added by qt3to4:
 #include <Q3VBoxLayout>
 
Index: src/frontends/q4banking/lib/widgets/qbjoblist.cpp
===================================================================
--- src/frontends/q4banking/lib/widgets/qbjoblist.cpp	(revision 1972)
+++ src/frontends/q4banking/lib/widgets/qbjoblist.cpp	(working copy)
@@ -17,8 +17,8 @@
 
 #include "qbjoblist.h"
 #include <assert.h>
-#include <Qt/qstring.h>
-#include <Qt/qwidget.h>
+#include <QtCore/qstring.h>
+#include <QtGui/qwidget.h>
 #include <gwenhywfar/debug.h>
 
 
Index: src/frontends/q4banking/lib/qbflagstaff.h
===================================================================
--- src/frontends/q4banking/lib/qbflagstaff.h	(revision 1972)
+++ src/frontends/q4banking/lib/qbflagstaff.h	(working copy)
@@ -16,7 +16,7 @@
 class QBFlagStaff;
 
 #include <q4banking/qbanking.h>
-#include <Qt/qobject.h>
+#include <QtCore/qobject.h>
 
 
 class Q4BANKING_API QBFlagStaff: public QObject {
Index: src/plugins/backends/aqofxconnect/ui/qt4/libs/cfgtabpageuserofx.cpp
--- src/plugins/backends/aqofxconnect/ui/qt4/libs/cfgtabpageuserofx.cpp	2010-01-15
+++ src/plugins/backends/aqofxconnect/ui/qt4/libs/cfgtabpageuserofx.cpp	2010-08-03
@@ -27,13 +27,13 @@
 #include <gwenhywfar/debug.h>
 #include <gwenhywfar/url.h>
 
-#include <qmessagebox.h>
-#include <qtimer.h>
-#include <qcheckbox.h>
-#include <qspinbox.h>
-#include <qpushbutton.h>
-#include <qlineedit.h>
-#include <qcombobox.h>
+#include <QtGui/qmessagebox.h>
+#include <QtCore/qtimer.h>
+#include <QtGui/qcheckbox.h>
+#include <QtGui/qspinbox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qcombobox.h>
 
 
 
Index: src/plugins/backends/aqofxconnect/ui/qt4/libs/cfgtabpageaccountofx.cpp
--- src/plugins/backends/aqofxconnect/ui/qt4/libs/cfgtabpageaccountofx.cpp 2010-01-15
+++ src/plugins/backends/aqofxconnect/ui/qt4/libs/cfgtabpageaccountofx.cpp 2010-08-03
@@ -22,11 +22,11 @@
 
 #include <q4banking/qbanking.h>
 
-#include <qmessagebox.h>
-#include <qtimer.h>
-#include <qcheckbox.h>
-#include <qspinbox.h>
-#include <q3filedialog.h>
+#include <QtGui/qmessagebox.h>
+#include <QtCore/qtimer.h>
+#include <QtGui/qcheckbox.h>
+#include <QtGui/qspinbox.h>
+#include <Qt3Support/q3filedialog.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/qt4_debug/qt4_debug.cpp
--- src/plugins/backends/aqhbci/ui/qt4/qt4_debug/qt4_debug.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/qt4_debug/qt4_debug.cpp	2010-08-03
@@ -15,10 +15,10 @@
 #endif
 
 
-#include <qapplication.h>
-#include <qtranslator.h>
-#include <qtextcodec.h>
-#include <qdir.h>
+#include <QtGui/qapplication.h>
+#include <QtCore/qtranslator.h>
+#include <QtCore/qtextcodec.h>
+#include <QtCore/qdir.h>
 
 #include <gwenhywfar/logger.h>
 #include <gwenhywfar/debug.h>
Index:src/plugins/backends/aqhbci/ui/qt4/qt4_debug/Makefile.in
--- src/plugins/backends/aqhbci/ui/qt4/qt4_debug/Makefile.in 2010-02-06
+++ src/plugins/backends/aqhbci/ui/qt4/qt4_debug/Makefile.in 2010-08-03
@@ -409,7 +409,7 @@
   $(aqbanking_internal_libs) \
   $(q4banking_internal_libs) \
   $(aqhbci_internal_libs) \
-  $(gwenhywfar_libs) $(qt3_libs) -lQtCore
+  $(gwenhywfar_libs) $(qt3_libs)
 
 CLEANFILES = qt4_debug.xml
 all: all-am
Index: src/plugins/backends/aqhbci/ui/qt4/libs/Makefile.in
--- src/plugins/backends/aqhbci/ui/qt4/libs/Makefile.in	2010-02-06
+++ src/plugins/backends/aqhbci/ui/qt4/libs/Makefile.in	2010-08-03
@@ -452,7 +452,7 @@
   $(aqhbci_internal_libs) \
   $(aqbanking_internal_libs) \
   $(q4banking_internal_libs) \
-  $(gwenhywfar_libs) $(qt4_libs) -lQt3Support
+  $(gwenhywfar_libs) $(qt4_libs)
 
 BUILT_SOURCES = $(nodist_aqhbci_la_SOURCES)
 CLEANFILES = $(nodist_aqhbci_la_SOURCES)
Index: src/plugins/backends/aqhbci/ui/qt4/libs/logmanager/logmanager.h
--- src/plugins/backends/aqhbci/ui/qt4/libs/logmanager/logmanager.h 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/logmanager/logmanager.h 2010-08-03
@@ -18,7 +18,7 @@
 #include "logmanager.ui.h"
 #include <gwenhywfar/msgengine.h>
 
-#include <QDialog>
+#include <QtGui/QDialog>
 
 #include <string>
 #include <list>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/logmanager/logmanager.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/logmanager/logmanager.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/logmanager/logmanager.cpp 2010-08-03
@@ -36,13 +36,13 @@
 #include <string.h>
 #include <errno.h>
 
-#include <qmessagebox.h>
-#include <qcombobox.h>
-#include <q3listview.h>
-#include <q3textbrowser.h>
-#include <qpushbutton.h>
-#include <q3filedialog.h>
-#include <qdir.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qcombobox.h>
+#include <Qt3Support/q3listview.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3filedialog.h>
+#include <QtCore/qdir.h>
 
 #include <q4banking/qbanking.h>
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/userwizard.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/userwizard.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/userwizard.cpp 2010-08-03
@@ -30,7 +30,7 @@
 #include <gwenhywfar/debug.h>
 #include <gwenhywfar/gui.h>
 
-#include <qmessagebox.h>
+#include <QtGui/qmessagebox.h>
 
 
 UserWizard::UserWizard(QBanking *qb, AB_PROVIDER *pro, QWidget *parent)
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/selectmode.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/selectmode.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/selectmode.cpp 2010-08-03
@@ -17,9 +17,9 @@
 
 #include "selectmode.h"
 
-#include <qradiobutton.h>
-#include <qtimer.h>
-#include <qlabel.h> // for qt4 setWordWrap(true)
+#include <QtGui/qradiobutton.h>
+#include <QtCore/qtimer.h>
+#include <QtGui/qlabel.h> // for qt4 setWordWrap(true)
 #include <gwenhywfar/debug.h>
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/cfgtabpageuserhbci.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/cfgtabpageuserhbci.cpp 2010-01-18
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/cfgtabpageuserhbci.cpp 2010-08-03
@@ -21,14 +21,14 @@
 #include <q4banking/qbanking.h>
 #include <q4banking/qbcfgtab.h>
 
-#include <qpushbutton.h>
-#include <qcombobox.h>
-#include <qlineedit.h>
-#include <qmessagebox.h>
-#include <q3listview.h>
-#include <qtimer.h>
-#include <q3groupbox.h>
-#include <qcheckbox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qmessagebox.h>
+#include <Qt3Support/q3listview.h>
+#include <QtCore/qtimer.h>
+#include <Qt3Support/q3groupbox.h>
+#include <QtGui/qcheckbox.h>
 
 #include <aqhbci/user.h>
 #include <aqhbci/provider.h>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/cfgtabpageaccounthbci.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/cfgtabpageaccounthbci.cpp 2010-01-18
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/cfgtabpageaccounthbci.cpp 2010-08-03
@@ -22,16 +22,16 @@
 
 #include <q4banking/qbanking.h>
 
-#include <qpushbutton.h>
-#include <qcombobox.h>
-#include <qlineedit.h>
-#include <qmessagebox.h>
-#include <q3listview.h>
-#include <qtimer.h>
-#include <q3groupbox.h>
-#include <qcheckbox.h>
-#include <qspinbox.h>
-#include <q3filedialog.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qmessagebox.h>
+#include <Qt3Support/q3listview.h>
+#include <QtCore/qtimer.h>
+#include <Qt3Support/q3groupbox.h>
+#include <QtGui/qcheckbox.h>
+#include <QtGui/qspinbox.h>
+#include <Qt3Support/q3filedialog.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_new2.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_new2.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_new2.cpp 2010-08-03
@@ -21,7 +21,7 @@
 
 #include <aqhbci/user.h>
 
-#include <qtimer.h>
+#include <QtCore/qtimer.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_new.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_new.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_new.cpp 2010-08-03
@@ -26,7 +26,7 @@
 
 #include <aqhbci/user.h>
 
-#include <qtimer.h>
+#include <QtCore/qtimer.h>
 
 
 WizardRdhNew::WizardRdhNew(QBanking *qb,
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_import.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_import.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_rdh_import.cpp 2010-08-03
@@ -26,7 +26,7 @@
 
 #include <aqhbci/user.h>
 
-#include <qtimer.h>
+#include <QtCore/qtimer.h>
 
 
 WizardRdhImport::WizardRdhImport(QBanking *qb,
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_ddv_import.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_ddv_import.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_ddv_import.cpp 2010-08-03
@@ -23,7 +23,7 @@
 
 #include <aqhbci/user.h>
 
-#include <qtimer.h>
+#include <QtCore/qtimer.h>
 
 
 WizardDdvImport::WizardDdvImport(QBanking *qb,
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_pintan_new.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_pintan_new.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/w_pintan_new.cpp 2010-08-03
@@ -23,7 +23,7 @@
 
 #include <aqhbci/user.h>
 
-#include <qtimer.h>
+#include <QtCore/qtimer.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getaccounts.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getaccounts.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getaccounts.cpp 2010-08-03
@@ -19,8 +19,8 @@
 #include "wizard.h"
 #include "actionwidget.h"
 
-#include <qpushbutton.h>
-#include <qmessagebox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
 
 #include <q4banking/qbanking.h>
 #include <aqhbci/provider.h>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_finished.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_finished.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_finished.cpp 2010-08-03
@@ -23,7 +23,7 @@
 
 #include <assert.h>
 
-#include <qlabel.h>
+#include <QtGui/qlabel.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_wait.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_wait.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_wait.cpp 2010-08-03
@@ -23,7 +23,7 @@
 
 #include <assert.h>
 
-#include <qlabel.h>
+#include <QtGui/qlabel.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getsysid.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getsysid.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getsysid.cpp 2010-08-03
@@ -24,8 +24,8 @@
 #include <gwenhywfar/debug.h>
 #include <gwenhywfar/gui.h>
 
-#include <qpushbutton.h>
-#include <qmessagebox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
 
 #include <assert.h>
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_useriniletter.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_useriniletter.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_useriniletter.cpp 2010-08-03
@@ -24,8 +24,8 @@
 
 #include <q4banking/qbanking.h>
 
-#include <qpushbutton.h>
-#include <q3textview.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3textview.h>
 
 #include <gwenhywfar/debug.h>
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_sendkeys.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_sendkeys.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_sendkeys.cpp 2010-08-03
@@ -19,7 +19,7 @@
 #include "wizard.h"
 #include "actionwidget.h"
 
-#include <qpushbutton.h>
+#include <QtGui/qpushbutton.h>
 
 #include <q4banking/qbanking.h>
 #include <aqhbci/provider.h>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_mkkeys.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_mkkeys.cpp 2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_mkkeys.cpp	2010-08-03
@@ -19,7 +19,7 @@
 #include "wizard.h"
 #include "actionwidget.h"
 
-#include <qpushbutton.h>
+#include <QtGui/qpushbutton.h>
 
 #include <q4banking/qbanking.h>
 #include <aqhbci/provider.h>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_bankiniletter.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_bankiniletter.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_bankiniletter.cpp	2010-08-03
@@ -23,8 +23,8 @@
 
 #include <q4banking/qbanking.h>
 
-#include <qpushbutton.h>
-#include <q3textview.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3textview.h>
 
 #include <gwenhywfar/debug.h>
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getkeys.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getkeys.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getkeys.cpp	2010-08-03
@@ -19,7 +19,7 @@
 #include "wizard.h"
 #include "actionwidget.h"
 
-#include <qpushbutton.h>
+#include <QtGui/qpushbutton.h>
 
 #include <q4banking/qbanking.h>
 #include <aqhbci/provider.h>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getcert.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getcert.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_getcert.cpp	2010-08-03
@@ -25,7 +25,7 @@
 
 #include <assert.h>
 
-#include <qlabel.h>
+#include <QtGui/qlabel.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_edituser.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_edituser.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_edituser.cpp	2010-08-03
@@ -19,7 +19,7 @@
 #include "wizard.h"
 #include "editctuser.h"
 
-#include <qlayout.h>
+#include <QtGui/qlayout.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_checkfile.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_checkfile.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_checkfile.cpp	2010-08-03
@@ -19,7 +19,7 @@
 #include "wizard.h"
 #include "actionwidget.h"
 
-#include <qpushbutton.h>
+#include <QtGui/qpushbutton.h>
 
 #include <q4banking/qbanking.h>
 #include <aqhbci/provider.h>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_createfile.H
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_createfile.H	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_createfile.H	2010-08-02
@@ -21,7 +21,7 @@
 #include "winfo.h"
 #include "a_selectfile.h"
 
-#include <qstring.h>
+#include <QtCore/qstring.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_selectfile.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_selectfile.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_selectfile.cpp	2010-08-02
@@ -21,11 +21,11 @@
 #include <aqhbci/provider.h>
 #include <gwenhywfar/debug.h>
 
-#include <qpushbutton.h>
-#include <qlineedit.h>
-#include <qfile.h>
-#include <q3filedialog.h>
-#include <qlabel.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qlineedit.h>
+#include <QtCore/qfile.h>
+#include <Qt3Support/q3filedialog.h>
+#include <QtGui/qlabel.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_selectfile.h
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_selectfile.h	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/a_selectfile.h	2010-08-02
@@ -18,7 +18,7 @@
 #include "winfo.h"
 #include "selectfile.ui.h"
 
-#include <qstring.h>
+#include <QtCore/qstring.h>
 
 
 class ActionSelectFile: public WizardAction {
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/iniletter.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/iniletter.cpp	2010-01-15 15:07:28.000000000 -0800
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/iniletter.cpp	2010-08-02 16:39:22.000000000 -0700
@@ -22,14 +22,11 @@
 
 #include <gwenhywfar/debug.h>
 
-#include <qmessagebox.h>
-#include <qstring.h>
-#include <qlabel.h>
-#include <qpushbutton.h>
-#include <q3textview.h>
-#include <qlabel.h> // for qt4 setWordWrap(true)
-
-
+#include <QtGui/qmessagebox.h>
+#include <QtCore/qstring.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3textview.h>
 
 IniLetter::IniLetter(bool useUserKey,
 		     AB_PROVIDER *pro,
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/iniletter.h
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/iniletter.h	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/iniletter.h	2010-08-02
@@ -19,7 +19,7 @@
 #include <aqbanking/provider.h>
 #include <aqbanking/user.h>
 
-#include <qstring.h>
+#include <QtCore/qstring.h>
 #include <string>
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/editctuser.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/editctuser.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/editctuser.cpp	2010-08-02
@@ -21,25 +21,25 @@
 #include <q4banking/qbanking.h>
 #include <q4banking/qbselectbank.h>
 //#include "selectcontext.h"
-#include <qradiobutton.h>
-#include <qpushbutton.h>
-#include <q3wizard.h>
-#include <qcombobox.h>
-#include <q3textbrowser.h>
-#include <qlabel.h>
-#include <q3groupbox.h>
-#include <qcheckbox.h>
-
-#include <qlineedit.h>
-#include <qmessagebox.h>
-#include <q3filedialog.h>
-#include <qpalette.h>
-#include <qbrush.h>
-#include <qpainter.h>
-#include <q3paintdevicemetrics.h>
-#include <qprinter.h>
-#include <q3simplerichtext.h>
-#include <q3textview.h>
+#include <QtGui/qradiobutton.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3wizard.h>
+#include <QtGui/qcombobox.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtGui/qlabel.h>
+#include <Qt3Support/q3groupbox.h>
+#include <QtGui/qcheckbox.h>
+
+#include <QtGui/qlineedit.h>
+#include <QtGui/qmessagebox.h>
+#include <Qt3Support/q3filedialog.h>
+#include <QtGui/qpalette.h>
+#include <QtGui/qbrush.h>
+#include <QtGui/qpainter.h>
+#include <Qt3Support/q3paintdevicemetrics.h>
+#include <QtGui/qprinter.h>
+#include <Qt3Support/q3simplerichtext.h>
+#include <Qt3Support/q3textview.h>
 
 #include <gwenhywfar/debug.h>
 #include <gwenhywfar/gui.h>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/actionwidget.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/actionwidget.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/actionwidget.cpp	2010-08-02
@@ -18,11 +18,11 @@
 
 #include <gwenhywfar/debug.h>
 
-#include <qmessagebox.h>
-#include <qstring.h>
-#include <qlabel.h>
-#include <q3textview.h>
-#include <qpushbutton.h>
+#include <QtGui/qmessagebox.h>
+#include <QtCore/qstring.h>
+#include <QtGui/qlabel.h>
+#include <Qt3Support/q3textview.h>
+#include <QtGui/qpushbutton.h>
 
 
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/wizard.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/wizard.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/wizard.cpp	2010-08-02
@@ -17,12 +17,12 @@
 
 #include "wizard.h"
 
-#include <qlabel.h>
-#include <qpushbutton.h>
-#include <q3listview.h>
-#include <qdatetime.h>
-#include <q3textbrowser.h>
-#include <qapplication.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3listview.h>
+#include <QtCore/qdatetime.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtGui/qapplication.h>
 
 #include <gwenhywfar/debug.h>
 
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/wizard.h
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/wizard.h	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/wizard.h	2010-08-02
@@ -25,8 +25,8 @@
 #include "waction.h"
 #include "winfo.h"
 
-#include <qstring.h>
-#include <q3wizard.h>
+#include <QtCore/qstring.h>
+#include <Qt3Support/q3wizard.h>
 #include <aqbanking/banking.h>
 
 #include <list>
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/waction.h
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/waction.h	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/waction.h	2010-08-02
@@ -1,5 +1,5 @@
 //Added by qt3to4:
-#include <Q3VBoxLayout>
+#include <Qt3Support/Q3VBoxLayout>
 /***************************************************************************
  $RCSfile$
  -------------------
@@ -22,8 +22,8 @@
 
 class Q3VBoxLayout;
 
-#include <qstring.h>
-#include <qwidget.h>
+#include <QtCore/qstring.h>
+#include <QtGui/qwidget.h>
 
 
 class WizardAction: public QWidget {
Index: src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/waction.cpp
--- src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/waction.cpp	2010-01-15
+++ src/plugins/backends/aqhbci/ui/qt4/libs/dialogs/wizard/waction.cpp	2010-08-02
@@ -17,10 +17,10 @@
 #include "waction.h"
 #include "wizard.h"
 
-#include <qwidget.h>
-#include <qlayout.h>
+#include <QtGui/qwidget.h>
+#include <QtGui/qlayout.h>
 //Added by qt3to4:
-#include <Q3VBoxLayout>
+#include <Qt3Support/Q3VBoxLayout>
 
 
 WizardAction::WizardAction(Wizard *w,
Index: src/aqbanking-4.2.4/src/frontends/q4banking/bin/helpbrowser/Makefile.in
--- src/frontends/q4banking/bin/helpbrowser/Makefile.in	2010-02-06
+++ src/frontends/q4banking/bin/helpbrowser/Makefile.in	2010-08-02
@@ -392,7 +392,7 @@
 
 q4b_help@Q4BANKING_SO_EFFECTIVE@_SOURCES = qb_help.cpp qbhelpbrowser.cpp
 q4b_help@Q4BANKING_SO_EFFECTIVE@_LDADD = \
-  $(qt4_libs) -lQtCore -lQt3Support \
+  $(qt4_libs) \
   $(gwenhywfar_libs)
 
 BUILT_SOURCES = $(MOC_FILES) $(nodist_libqbhelp_la_SOURCES) $(nodist_noinst_HEADERS)
Index: src/frontends/q4banking/bin/helpbrowser/qbhelpbrowser.cpp
--- src/frontends/q4banking/bin/helpbrowser/qbhelpbrowser.cpp	2010-01-15
+++ src/frontends/q4banking/bin/helpbrowser/qbhelpbrowser.cpp	2010-08-02
@@ -15,34 +15,34 @@
 #endif
 
 #include "qbhelpbrowser.h"
-#include <qstatusbar.h>
-#include <qpixmap.h>
-#include <q3popupmenu.h>
-#include <qmenubar.h>
-#include <q3toolbar.h>
-#include <qtoolbutton.h>
-#include <qicon.h>
-#include <qfile.h>
-#include <q3textstream.h>
-#include <q3stylesheet.h>
-#include <qmessagebox.h>
-#include <q3filedialog.h>
-#include <qapplication.h>
-#include <qcombobox.h>
-#include <qevent.h>
-#include <qlineedit.h>
-#include <qobject.h>
-#include <qfileinfo.h>
-#include <qfile.h>
-#include <qdatastream.h>
-#include <qprinter.h>
-#include <q3simplerichtext.h>
-#include <qpainter.h>
-#include <q3paintdevicemetrics.h>
+#include <QtGui/qstatusbar.h>
+#include <QtGui/qpixmap.h>
+#include <Qt3Support/q3popupmenu.h>
+#include <QtGui/qmenubar.h>
+#include <Qt3Support/q3toolbar.h>
+#include <QtGui/qtoolbutton.h>
+#include <QtGui/qicon.h>
+#include <QtCore/qfile.h>
+#include <Qt3Support/q3textstream.h>
+#include <Qt3Support/q3stylesheet.h>
+#include <QtGui/qmessagebox.h>
+#include <Qt3Support/q3filedialog.h>
+#include <QtGui/qapplication.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qevent.h>
+#include <QtGui/qlineedit.h>
+#include <QtCore/qobject.h>
+#include <QtCore/qfileinfo.h>
+#include <QtCore/qfile.h>
+#include <QtCore/qdatastream.h>
+#include <QtGui/qprinter.h>
+#include <Qt3Support/q3simplerichtext.h>
+#include <QtGui/qpainter.h>
+#include <Qt3Support/q3paintdevicemetrics.h>
 
-#include <q3textbrowser.h>
+#include <Qt3Support/q3textbrowser.h>
 //Added by qt3to4:
-#include <Q3Frame>
+#include <Qt3Support/Q3Frame>
 
 #include <ctype.h>
 
Index: src/frontends/q4banking/bin/helpbrowser/qbhelpbrowser.h
--- src/frontends/q4banking/bin/helpbrowser/qbhelpbrowser.h	2010-01-15
+++ src/frontends/q4banking/bin/helpbrowser/qbhelpbrowser.h	2010-08-02
@@ -13,17 +13,17 @@
 #ifndef QBHELPWINDOW_H
 #define QBHELPWINDOW_H
 
-#include <q3mainwindow.h>
-#include <q3textbrowser.h>
-#include <qstringlist.h>
-#include <qmap.h>
-#include <qdir.h>
-#include <qstringlist.h>
+#include <Qt3Support/q3mainwindow.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtCore/qstringlist.h>
+#include <QtCore/qmap.h>
+#include <QtCore/qdir.h>
+#include <QtCore/qstringlist.h>
 
 #include "qbhelpbrowser.ui.h"
 #include <q4banking/qbanking.h>
 //Added by qt3to4:
-#include <Q3PopupMenu>
+#include <Qt3Support/Q3PopupMenu>
 
 
 class QComboBox;
Index: src/frontends/q4banking/bin/helpbrowser/qb_help.cpp
--- src/frontends/q4banking/bin/helpbrowser/qb_help.cpp	2010-01-15
+++ src/frontends/q4banking/bin/helpbrowser/qb_help.cpp	2010-08-02
@@ -15,9 +15,11 @@
 #endif
 
 
-#include <qapplication.h>
-#include <qtranslator.h>
-#include <qtextcodec.h>
+#include <QtGui/qapplication.h>
+#include <QtCore/qtranslator.h>
+#include <QtCore/qtextcodec.h>
+#include <QtCore/qstringlist.h>
+#include <QtGui/qmessagebox.h>
 
 #include <gwenhywfar/logger.h>
 #include <gwenhywfar/debug.h>
@@ -26,11 +28,6 @@
 
 #include "qbhelpbrowser.h"
 
-#include <qapplication.h>
-#include <qstringlist.h>
-#include <qtextcodec.h>
-#include <qtranslator.h>
-#include <qmessagebox.h>
 
 
 #ifdef OS_WIN32
Index: src/frontends/q4banking/bin/wizard/qt4_wizard.cpp
--- src/frontends/q4banking/bin/wizard/qt4_wizard.cpp	2010-01-15
+++ src/frontends/q4banking/bin/wizard/qt4_wizard.cpp	2010-08-02
@@ -15,7 +15,7 @@
 #endif
 
 
-#include <qapplication.h>
+#include <QtGui/qapplication.h>
 
 #include <gwenhywfar/logger.h>
 #include <gwenhywfar/debug.h>
Index: src/frontends/q4banking/lib/checklib.cpp
--- src/frontends/q4banking/lib/checklib.cpp	2010-01-15
+++ src/frontends/q4banking/lib/checklib.cpp	2010-08-02
@@ -1,15 +1,15 @@
 #include <q4banking/qbanking.h>
 
-#include <qapplication.h>
-#include <qmessagebox.h>
-#include <qtranslator.h>
-#include <qtextcodec.h>
-#include <qregexp.h>
+#include <QtGui/qapplication.h>
+#include <QtGui/qmessagebox.h>
+#include <QtCore/qtranslator.h>
+#include <Qtcore/qtextcodec.h>
+#include <QtCore/qregexp.h>
 
 #include <gwenhywfar/debug.h>
 #include <gwenhywfar/gwenhywfar.h>
 
-#include <qdir.h>
+#include <QtCore/qdir.h>
 
 
 int main(int argc, char *argv[]){
Index: src/frontends/q4banking/lib/testlib.cpp
--- src/frontends/q4banking/lib/testlib.cpp	2010-01-15
+++ src/frontends/q4banking/lib/testlib.cpp	2010-08-02
@@ -1,7 +1,7 @@
 #include <q4banking/qbanking.h>
 
-#include <qapplication.h>
-#include <qmessagebox.h>
+#include <QtGui/qapplication.h>
+#include <QtGui/qmessagebox.h>
 
 #include <gwenhywfar/debug.h>
 #include <gwenhywfar/gwenhywfar.h>
Index: src/frontends/q4banking/lib/Makefile.in
--- src/frontends/q4banking/lib/Makefile.in	2010-02-06
+++ src/frontends/q4banking/lib/Makefile.in	2010-08-03
@@ -493,7 +493,7 @@
  dialogs/libdialogs.la \
  views/libviews.la \
  gui/libqgui.la \
- $(i18n_libs) $(aqbanking_internal_libs) $(gwenhywfar_libs) $(qt4_libs) -lQt3Support
+ $(i18n_libs) $(aqbanking_internal_libs) $(gwenhywfar_libs) $(qt4_libs)
 
 libq4banking_la_LDFLAGS = -no-undefined -version-info \
 	@Q4BANKING_SO_CURRENT@:@Q4BANKING_SO_REVISION@:@Q4BANKING_SO_AGE@ \
@@ -504,9 +504,9 @@
 	dialogs/libdialogs.la views/libviews.la gui/libqgui.la \
 	$(aqbanking_internal_libs) $(am__append_1)
 testlib_SOURCES = testlib.cpp
-testlib_LDADD = libq4banking.la $(aqbanking_internal_libs) $(gwenhywfar_libs) $(qt4_libs) -lQtCore -lQt3Support
+testlib_LDADD = libq4banking.la $(aqbanking_internal_libs) $(gwenhywfar_libs) $(qt4_libs)
 checklib_SOURCES = checklib.cpp
-checklib_LDADD = libq4banking.la $(aqbanking_internal_libs) $(gwenhywfar_libs) $(qt4_libs) -lQtCore -lQt3Support
+checklib_LDADD = libq4banking.la $(aqbanking_internal_libs) $(gwenhywfar_libs) $(qt4_libs)
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
Index: src/frontends/q4banking/bin/wizard/Makefile.in
--- src/frontends/q4banking/bin/wizard/Makefile.in	2010-02-06
+++ src/frontends/q4banking/bin/wizard/Makefile.in	2010-08-03
@@ -410,7 +410,7 @@
 qt4_wizard_LDADD = \
   $(q4banking_internal_libs) \
   $(aqbanking_internal_libs) \
-  $(gwenhywfar_libs) $(qt4_libs) -lQtCore
+  $(gwenhywfar_libs) $(qt4_libs)
 
 CLEANFILES = qt4_wizard.xml
 all: all-am
Index: src/frontends/q4banking/lib/qbgui.cpp
--- src/frontends/q4banking/lib/qbgui.cpp	2010-01-15
+++ src/frontends/q4banking/lib/qbgui.cpp	2010-08-02
@@ -20,7 +20,7 @@
 #include "qbprintdialog.h"
 
 
-#include <qwidget.h>
+#include <QtGui/qwidget.h>
 
 #include <aqbanking/abgui.h>
 
Index: src/frontends/q4banking/lib/qbanking.cpp
--- src/frontends/q4banking/lib/qbanking.cpp	2010-01-15
+++ src/frontends/q4banking/lib/qbanking.cpp	2010-08-02
@@ -26,22 +26,22 @@
 #include "qbcfgmodule.h"
 #include "qbcfgtabsettings.h"
 //Added by qt3to4:
-#include <Q3CString>
+#include <Qt3Support/Q3CString>
 
 #include <aqbanking/jobgetbalance.h>
 #include <aqbanking/jobgettransactions.h>
 
 #include <assert.h>
-#include <qstring.h>
-#include <qmessagebox.h>
-#include <qinputdialog.h>
-#include <qapplication.h>
-#include <qdatetime.h>
-#include <qwidget.h>
-#include <qtranslator.h>
-#include <qtextcodec.h>
-#include <q3process.h>
-#include <qdir.h>
+#include <QtCore/qstring.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qinputdialog.h>
+#include <QtGui/qapplication.h>
+#include <QtCore/qdatetime.h>
+#include <QtGui/qwidget.h>
+#include <QtCore/qtranslator.h>
+#include <QtCore/qtextcodec.h>
+#include <Qt3Support/q3process.h>
+#include <QtCore/qdir.h>
 
 #include <gwenhywfar/debug.h>
 #include <gwenhywfar/libloader.h>
Index: src/frontends/q4banking/lib/gui/qguisimplebox.cpp
--- src/frontends/q4banking/lib/gui/qguisimplebox.cpp	2010-01-15
+++ src/frontends/q4banking/lib/gui/qguisimplebox.cpp	2010-08-02
@@ -15,8 +15,8 @@
 #endif
 
 
-#include <qlabel.h>
-#include <qtimer.h>
+#include <QtGui/qlabel.h>
+#include <QtCore/qtimer.h>
 
 #include "qguisimplebox.h"
 
Index: src/frontends/q4banking/lib/gui/qguiprogresswidget.cpp
--- src/frontends/q4banking/lib/gui/qguiprogresswidget.cpp	2010-01-15
+++ src/frontends/q4banking/lib/gui/qguiprogresswidget.cpp	2010-08-02
@@ -17,16 +17,16 @@
 #include "qguiprogresswidget.h"
 #include "qguiprogress.h"
 
-#include <qlabel.h>
-#include <q3progressbar.h>
-#include <qapplication.h>
-#include <qpushbutton.h>
-#include <qdatetime.h>
-#include <q3textbrowser.h>
-#include <q3progressbar.h>
-#include <q3groupbox.h>
+#include <QtGui/qlabel.h>
+#include <Qt3Support/q3progressbar.h>
+#include <QtGui/qapplication.h>
+#include <QtGui/qpushbutton.h>
+#include <QtCore/qdatetime.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <Qt3Support/q3progressbar.h>
+#include <Qt3Support/q3groupbox.h>
 //Added by qt3to4:
-#include <QCloseEvent>
+#include <QtGui/QCloseEvent>
 
 #include <gwenhywfar/debug.h>
 
Index:src/frontends/q4banking/lib/gui/qguiinputbox.cpp
--- src/frontends/q4banking/lib/gui/qguiinputbox.cpp	2010-01-15
+++ src/frontends/q4banking/lib/gui/qguiinputbox.cpp	2010-08-02
@@ -15,21 +15,21 @@
 #endif
 
 
-#include <qlabel.h>
-#include <qapplication.h>
-#include <qpushbutton.h>
-#include <q3textbrowser.h>
-#include <qlineedit.h>
-#include <q3simplerichtext.h>
-#include <qvalidator.h>
-#include <qlayout.h>
-#include <qtimer.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qapplication.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtGui/qlineedit.h>
+#include <Qt3Support/q3simplerichtext.h>
+#include <QtGui/qvalidator.h>
+#include <QtGui/qlayout.h>
+#include <QtCore/qtimer.h>
 //Added by qt3to4:
-#include <Q3GridLayout>
-#include <Q3BoxLayout>
-#include <Q3HBoxLayout>
-#include <Q3VBoxLayout>
-#include <Q3Frame>
+#include <Qt3Support/Q3GridLayout>
+#include <Qt3Support/Q3BoxLayout>
+#include <Qt3Support/Q3HBoxLayout>
+#include <Qt3Support/Q3VBoxLayout>
+#include <Qt3Support/Q3Frame>
 
 #include <gwenhywfar/debug.h>
 #include <gwenhywfar/gui.h>
Index:  src/frontends/q4banking/lib/gui/qgui.cpp
--- src/frontends/q4banking/lib/gui/qgui.cpp	2010-01-15
+++ src/frontends/q4banking/lib/gui/qgui.cpp	2010-08-02
@@ -21,12 +21,12 @@
 #include "qguiinputbox.h"
 #include "qguisimplebox.h"
 //Added by qt3to4:
-#include <Q3CString>
+#include <Qt3Support/Q3CString>
 
 #include <gwenhywfar/debug.h>
 
-#include <qmessagebox.h>
-#include <qapplication.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qapplication.h>
 
 #include <assert.h>
 
Index: src/frontends/q4banking/lib/dialogs/qbmapaccount.cpp
--- src/frontends/q4banking/lib/dialogs/qbmapaccount.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbmapaccount.cpp	2010-08-02
@@ -21,13 +21,13 @@
 #include "qbaccountlist.h"
 
 // QT includes
-#include <qlabel.h>
-#include <qpushbutton.h>
-#include <qmessagebox.h>
-#include <qlayout.h>
-#include <qpushbutton.h>
-#include <q3groupbox.h>
-#include <qlineedit.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qlayout.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3groupbox.h>
+#include <QtGui/qlineedit.h>
 
 
 
Index: src/frontends/q4banking/lib/gui/qguiinputbox.h
--- src/frontends/q4banking/lib/gui/qguiinputbox.h	2010-01-15
+++ src/frontends/q4banking/lib/gui/qguiinputbox.h	2010-08-02
@@ -16,8 +16,8 @@
 
 #include <gwenhywfar/types.h>
 
-#include <qdialog.h>
-#include <qvalidator.h>
+#include <QtGui/qdialog.h>
+#include <QtGui/qvalidator.h>
 
 
 class QLineEdit;
Index: src/frontends/q4banking/lib/gui/qguiprogresswidget.h
--- src/frontends/q4banking/lib/gui/qguiprogresswidget.h	2010-01-15
+++ src/frontends/q4banking/lib/gui/qguiprogresswidget.h	2010-08-02
@@ -16,7 +16,7 @@
 
 #include "qguiprogresswidget.ui.h"
 //Added by qt3to4:
-#include <QCloseEvent>
+#include <QtGui/QCloseEvent>
 #include <gwenhywfar/types.h>
 #include <gwenhywfar/gui.h>
 
Index: src/aqbanking-4.2.4/src/frontends/q4banking/lib/gui/qguiprogress.h
--- src/frontends/q4banking/lib/gui/qguiprogress.h	2010-01-15
+++ src/frontends/q4banking/lib/gui/qguiprogress.h	2010-08-02
@@ -18,7 +18,7 @@
 
 #include <time.h>
 
-#include <qstring.h>
+#include <QtCore/qstring.h>
 
 
 class QGuiProgressWidget;
Index: src/frontends/q4banking/lib/widgets/qbcfgtab.h
--- src/frontends/q4banking/lib/widgets/qbcfgtab.h	2010-01-15
+++ src/frontends/q4banking/lib/widgets/qbcfgtab.h	2010-08-02
@@ -20,7 +20,7 @@
 #include <q4banking/qbanking.h>
 
 #include "qbcfgtab.ui.h"
-#include <qstring.h>
+#include <QtCore/qstring.h>
 
 
 class QBanking;
Index: src/frontends/q4banking/lib/views/qbbankselector.cpp
--- src/frontends/q4banking/lib/views/qbbankselector.cpp	2010-01-15
+++ src/frontends/q4banking/lib/views/qbbankselector.cpp	2010-08-02
@@ -17,7 +17,7 @@
 
 #include "qbbankselector.h"
 #include "qbselectbank.h"
-#include <qlineedit.h>
+#include <QtGui/qlineedit.h>
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbselectbackend.cpp
--- src/frontends/q4banking/lib/dialogs/qbselectbackend.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbselectbackend.cpp	2010-08-02
@@ -22,10 +22,10 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <qcombobox.h>
-#include <qpushbutton.h>
-#include <q3textbrowser.h>
-#include <qlabel.h> // for qt4 setWordWrap(true)
+#include <QtGui/qcombobox.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtGui/qlabel.h> // for qt4 setWordWrap(true)
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbeditaccount.cpp
--- src/frontends/q4banking/lib/dialogs/qbeditaccount.cpp	2010-01-24
+++ src/frontends/q4banking/lib/dialogs/qbeditaccount.cpp	2010-08-02
@@ -21,7 +21,7 @@
 #include "qbanking.h"
 
 // QT includes
-#include <qmessagebox.h>
+#include <QtGui/qmessagebox.h>
 
 // Gwenhywfar includes
 #include <gwenhywfar/debug.h>
Index: src/frontends/q4banking/lib/dialogs/qbedituser.cpp
--- src/frontends/q4banking/lib/dialogs/qbedituser.cpp	2010-01-24
+++ src/frontends/q4banking/lib/dialogs/qbedituser.cpp	2010-08-02
@@ -21,7 +21,7 @@
 #include "qbanking.h"
 
 // QT includes
-#include <qmessagebox.h>
+#include <QtGui/qmessagebox.h>
 
 // Gwenhywfar includes
 #include <gwenhywfar/debug.h>
Index: src/frontends/q4banking/lib/dialogs/qbcfgtabpagebackends.cpp
--- src/frontends/q4banking/lib/dialogs/qbcfgtabpagebackends.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbcfgtabpagebackends.cpp	2010-08-02
@@ -24,11 +24,11 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <qlineedit.h>
-#include <qcombobox.h>
-#include <qpushbutton.h>
-#include <qmessagebox.h>
-#include <qlayout.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qlayout.h>
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbcfgtabpageusers.cpp
--- src/frontends/q4banking/lib/dialogs/qbcfgtabpageusers.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbcfgtabpageusers.cpp	2010-08-02
@@ -27,12 +27,12 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <qlineedit.h>
-#include <qcombobox.h>
-#include <qpushbutton.h>
-#include <qmessagebox.h>
-#include <qlayout.h>
-#include <qtextcodec.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qlayout.h>
+#include <QtCore/qtextcodec.h>
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbimporter.cpp
--- src/frontends/q4banking/lib/dialogs/qbimporter.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbimporter.cpp	2010-08-02
@@ -27,17 +27,17 @@
 #include <gwenhywfar/iomanager.h>
 
 // QT includes
-#include <q3listview.h>
-#include <qpushbutton.h>
-#include <q3textbrowser.h>
-#include <qmessagebox.h>
-#include <qapplication.h>
-#include <q3progressbar.h>
-#include <qfile.h>
-#include <q3filedialog.h>
-#include <qcombobox.h>
-#include <qlabel.h>
-#include <qlineedit.h>
+#include <Qt3Support/q3listview.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qapplication.h>
+#include <Qt3Support/q3progressbar.h>
+#include <QtCore/qfile.h>
+#include <Qt3Support/q3filedialog.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qlineedit.h>
 
 // C includes
 #include <sys/types.h>
Index: src/frontends/q4banking/lib/dialogs/qbedituser.h
--- src/frontends/q4banking/lib/dialogs/qbedituser.h	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbedituser.h	2010-08-02
@@ -16,7 +16,7 @@
 
 #include <q4banking/qbcfgtab.h>
 
-#include <qstring.h>
+#include <QtCore/qstring.h>
 
 
 class QBanking;
Index: src/frontends/q4banking/lib/dialogs/qbcfgtabpageaccounts.cpp
--- src/frontends/q4banking/lib/dialogs/qbcfgtabpageaccounts.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbcfgtabpageaccounts.cpp	2010-08-02
@@ -27,12 +27,12 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <qlineedit.h>
-#include <qcombobox.h>
-#include <qpushbutton.h>
-#include <qmessagebox.h>
-#include <qlayout.h>
-#include <qtextcodec.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qlayout.h>
+#include <QtCore/qtextcodec.h>
 
 
 
Index: src/frontends/q4banking/lib/qbcfgmodule.h
--- src/frontends/q4banking/lib/qbcfgmodule.h	2010-01-15 15:07:21.000000000 -0800
+++ src/frontends/q4banking/lib/qbcfgmodule.h	2010-08-02 15:46:29.000000000 -0700
@@ -17,7 +17,7 @@
 #include <aqbanking/banking.h>
 #include <gwenhywfar/plugin.h>
 
-#include <qstring.h>
+#include <QtCore/qstring.h>
 
 
 #define Q4BCFGMODULE_FLAGS_CAN_CREATE_USER    0x00000001
Index: src/frontends/q4banking/lib/dialogs/qbcfgtabpageaccountgen.cpp
--- src/frontends/q4banking/lib/dialogs/qbcfgtabpageaccountgen.cpp	2010-01-18
+++ src/frontends/q4banking/lib/dialogs/qbcfgtabpageaccountgen.cpp	2010-08-02
@@ -22,12 +22,12 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <qlineedit.h>
-#include <qcombobox.h>
-#include <qpushbutton.h>
-#include <qmessagebox.h>
-#include <qlayout.h>
-#include <qcheckbox.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qlayout.h>
+#include <QtGui/qcheckbox.h>
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbcfgtabpageusergen.cpp
--- src/frontends/q4banking/lib/dialogs/qbcfgtabpageusergen.cpp	2010-01-18
+++ src/frontends/q4banking/lib/dialogs/qbcfgtabpageusergen.cpp	2010-08-02
@@ -22,12 +22,12 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <qlineedit.h>
-#include <qcombobox.h>
-#include <qpushbutton.h>
-#include <qmessagebox.h>
-#include <qlabel.h>
-#include <qlocale.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qcombobox.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qlabel.h>
+#include <QtCore/qlocale.h>
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbprintdialog.h
--- src/frontends/q4banking/lib/dialogs/qbprintdialog.h	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbprintdialog.h	2010-08-02
@@ -17,7 +17,7 @@
 
 #include "qbprintdialog.ui.h"
 
-#include <qstring.h>
+#include <QtCore/qstring.h>
 
 class QPrinter;
 
Index: src/frontends/q4banking/lib/dialogs/qbprintdialog.cpp
--- src/frontends/q4banking/lib/dialogs/qbprintdialog.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbprintdialog.cpp	2010-08-02
@@ -25,15 +25,15 @@
 #include <gwenhywfar/gui.h>
 
 // QT includes
-#include <qlabel.h>
-#include <q3textbrowser.h>
-#include <qpushbutton.h>
-#include <qprinter.h>
-#include <qpainter.h>
-#include <q3paintdevicemetrics.h>
-#include <q3simplerichtext.h>
-#include <qmessagebox.h>
-#include <qfontdialog.h>
+#include <QtGui/qlabel.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qprinter.h>
+#include <QtGui/qpainter.h>
+#include <Qt3Support/q3paintdevicemetrics.h>
+#include <Qt3Support/q3simplerichtext.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qfontdialog.h>
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbselectfromlist.cpp
--- src/frontends/q4banking/lib/dialogs/qbselectfromlist.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbselectfromlist.cpp	2010-08-02
@@ -22,10 +22,10 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <qlabel.h>
-#include <qpushbutton.h>
-#include <q3groupbox.h>
-#include <q3listview.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3groupbox.h>
+#include <Qt3Support/q3listview.h>
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbselectbank.cpp
--- src/frontends/q4banking/lib/dialogs/qbselectbank.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbselectbank.cpp	2010-08-02
@@ -27,12 +27,12 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <q3listview.h>
-#include <qlineedit.h>
-#include <qlabel.h>
-#include <qmessagebox.h>
-#include <qcheckbox.h>
-#include <qpushbutton.h>
+#include <Qt3Support/q3listview.h>
+#include <QtGui/qlineedit.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qmessagebox.h>
+#include <QtGui/qcheckbox.h>
+#include <QtGui/qpushbutton.h>
 
 // C++ includes
 #include <string>
Index: src/frontends/q4banking/lib/dialogs/qbpickstartdate.cpp
--- src/frontends/q4banking/lib/dialogs/qbpickstartdate.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbpickstartdate.cpp	2010-08-02
@@ -21,11 +21,11 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <q3datetimeedit.h>
-#include <qradiobutton.h>
-#include <qlabel.h>
-#include <qpushbutton.h>
-#include <q3buttongroup.h>
+#include <Qt3Support/q3datetimeedit.h>
+#include <QtGui/qradiobutton.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qpushbutton.h>
+#include <Qt3Support/q3buttongroup.h>
 
 
 
Index: src/frontends/q4banking/lib/dialogs/qbpickstartdate.h
--- src/frontends/q4banking/lib/dialogs/qbpickstartdate.h	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbpickstartdate.h	2010-08-02
@@ -15,7 +15,7 @@
 
 #include <q4banking/qbpickstartdate.ui.h>
 #include <q4banking/qbanking.h>
-#include <qdatetime.h>
+#include <QtCore/qdatetime.h>
 
 
 class QBanking;
Index: src/frontends/q4banking/lib/dialogs/qbselectfromlist.h
--- src/frontends/q4banking/lib/dialogs/qbselectfromlist.h	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbselectfromlist.h	2010-08-02
@@ -16,7 +16,7 @@
 class QBanking;
 
 
-#include <qstringlist.h>
+#include <QtCore/qstringlist.h>
 
 #include <q4banking/qbselectfromlist.ui.h>
 #include <q4banking/qbanking.h>
Index: src/frontends/q4banking/lib/dialogs/qbprocesswatcher.cpp
--- src/frontends/q4banking/lib/dialogs/qbprocesswatcher.cpp	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbprocesswatcher.cpp	2010-08-02
@@ -22,11 +22,11 @@
 #include <gwenhywfar/debug.h>
 
 // QT includes
-#include <q3process.h>
-#include <qapplication.h>
-#include <qlabel.h>
-#include <qpushbutton.h>
-#include <qmessagebox.h>
+#include <Qt3Support/q3process.h>
+#include <QtGui/qapplication.h>
+#include <QtGui/qlabel.h>
+#include <QtGui/qpushbutton.h>
+#include <QtGui/qmessagebox.h>
 
 
 
Index: src/frontends/q4banking/lib/widgets/qbcfgtabpage.cpp
--- src/frontends/q4banking/lib/widgets/qbcfgtabpage.cpp	2010-01-18
+++ src/frontends/q4banking/lib/widgets/qbcfgtabpage.cpp	2010-08-02
@@ -17,17 +17,17 @@
 
 #include "qbcfgtabpage.h"
 //Added by qt3to4:
-#include <Q3VBoxLayout>
+#include <Qt3Support/Q3VBoxLayout>
 
 #include <gwenhywfar/debug.h>
 
 
-#include <qlayout.h>
+#include <QtGui/qlayout.h>
 #if (QT_VERSION >= 0x040100)
 // qt-4.1.0 errorneously forgets to add this include
-#  include <QComboBox>
+#  include <QtGui/QComboBox>
 #endif
-#include <qcombobox.h>
+#include <QtGui/qcombobox.h>
 
 
 QBCfgTabPage::QBCfgTabPage(QBanking *qb,
Index: src/frontends/q4banking/lib/dialogs/qbprocesswatcher.h
--- src/frontends/q4banking/lib/dialogs/qbprocesswatcher.h	2010-01-15
+++ src/frontends/q4banking/lib/dialogs/qbprocesswatcher.h	2010-08-02
@@ -17,7 +17,7 @@
 #include <q4banking/qbprocesswatcher.ui.h>
 #include <q4banking/qbanking.h>
 
-#include <qstring.h>
+#include <QtCore/qstring.h>
 #include <time.h>
 
 class Q3Process;
Index: src/frontends/q4banking/lib/widgets/qbuserlist.cpp
--- src/frontends/q4banking/lib/widgets/qbuserlist.cpp	2010-01-15
+++ src/frontends/q4banking/lib/widgets/qbuserlist.cpp	2010-08-02
@@ -17,7 +17,7 @@
 
 #include "qbuserlist.h"
 #include <assert.h>
-#include <qstring.h>
+#include <QtCore/qstring.h>
 
 #include <gwenhywfar/debug.h>
 
Index: src/frontends/q4banking/lib/widgets/qbcfgtab.cpp
--- src/frontends/q4banking/lib/widgets/qbcfgtab.cpp	2010-01-15
+++ src/frontends/q4banking/lib/widgets/qbcfgtab.cpp	2010-08-02
@@ -18,9 +18,9 @@
 #include "qbcfgtab.h"
 #include "qbcfgtabpage.h"
 
-#include <qtabwidget.h>
-#include <q3textbrowser.h>
-#include <qpushbutton.h>
+#include <QtGui/qtabwidget.h>
+#include <Qt3Support/q3textbrowser.h>
+#include <QtGui/qpushbutton.h>
 
 #include <gwenhywfar/debug.h>
 
Index: src/frontends/q4banking/lib/widgets/qbcfgtabpage.h
--- src/frontends/q4banking/lib/widgets/qbcfgtabpage.h	2010-08-02
+++ src/frontends/q4banking/lib/widgets/qbcfgtabpage.h	2010-08-02
@@ -21,7 +21,7 @@
 #include <QtGui/qwidget.h>
 #include <QtCore/qstring.h>
 //Added by qt3to4:
-#include <Q3VBoxLayout>
+#include <Qt3Support/Q3VBoxLayout>
 
 
 
Index: src/frontends/q4banking/lib/gui/qgui.h
--- src/frontends/q4banking/lib/gui/qgui.h	2010-01-15
+++ src/frontends/q4banking/lib/gui/qgui.h	2010-08-02
@@ -10,7 +10,7 @@
 
 #include <q4banking/cppgui.h>
 
-#include <Qt/qstring.h>
+#include <QtCore/qstring.h>
 #include <string>
 
 class QBanking;
