--- a/src/frontends/qbanking/lib/dialogs/qbcfgtabpageaccountgen.ui	2010-06-13 15:20:48.000000000 -0700
+++ b/src/frontends/qbanking/lib/dialogs/qbcfgtabpageaccountgen.ui	2010-06-13 15:22:53.000000000 -0700
@@ -187,6 +187,7 @@
                             <property name="title">
                                 <string>Bank Settings</string>
                             </property>
+			    <grid>
                             <widget class="QLineEdit">
                                 <property name="name">
                                     <cstring>bankNameEdit</cstring>
@@ -319,6 +320,7 @@
                                     <string>...</string>
                                 </property>
                             </widget>
+			    </grid>
                         </widget>
                         <spacer>
                             <property name="name">
