--- a/src/frontends/qbanking/lib/dialogs/qbcfgtabpageaccountgen.ui	2010-06-13 16:40:41.000000000 -0700
+++ b/src/frontends/qbanking/lib/dialogs/qbcfgtabpageaccountgen.ui	2010-06-13 16:50:05.000000000 -0700
@@ -187,7 +187,11 @@
                             <property name="title">
                                 <string>Bank Settings</string>
                             </property>
-                            <widget class="QLineEdit">
+			    <grid>
+			      <property name="name">
+				<cstring>unnamed</cstring>
+			      </property>
+                           <widget class="QLineEdit" row="2" column="1">
                                 <property name="name">
                                     <cstring>bankNameEdit</cstring>
                                 </property>
@@ -200,7 +204,7 @@
                                     </rect>
                                 </property>
                             </widget>
-                            <widget class="QLineEdit">
+                            <widget class="QLineEdit" row="1" column="1">
                                 <property name="name">
                                     <cstring>bankIdEdit</cstring>
                                 </property>
@@ -213,7 +217,7 @@
                                     </rect>
                                 </property>
                             </widget>
-                            <widget class="QComboBox">
+                            <widget class="QComboBox" row="0" column="1">
                                 <property name="name">
                                     <cstring>countryCombo</cstring>
                                 </property>
@@ -226,7 +230,7 @@
                                     </rect>
                                 </property>
                             </widget>
-                            <widget class="QLabel">
+                            <widget class="QLabel" row="3" column="0">
                                 <property name="name">
                                     <cstring>textLabel1_4_3</cstring>
                                 </property>
@@ -242,7 +246,7 @@
                                     <string>BIC</string>
                                 </property>
                             </widget>
-                            <widget class="QLabel">
+                            <widget class="QLabel" row="2" column="0">
                                 <property name="name">
                                     <cstring>textLabel1_4_2</cstring>
                                 </property>
@@ -258,7 +262,7 @@
                                     <string>Name</string>
                                 </property>
                             </widget>
-                            <widget class="QLineEdit">
+                            <widget class="QLineEdit" row="3" column="1">
                                 <property name="name">
                                     <cstring>bicEdit</cstring>
                                 </property>
@@ -271,7 +275,7 @@
                                     </rect>
                                 </property>
                             </widget>
-                            <widget class="QLabel">
+                            <widget class="QLabel" row="1" column="0">
                                 <property name="name">
                                     <cstring>textLabel1_4</cstring>
                                 </property>
@@ -287,7 +291,7 @@
                                     <string>Code</string>
                                 </property>
                             </widget>
-                            <widget class="QLabel">
+                            <widget class="QLabel" row="0" column="0">
                                 <property name="name">
                                     <cstring>textLabel2_2</cstring>
                                 </property>
@@ -303,7 +307,7 @@
                                     <string>Country</string>
                                 </property>
                             </widget>
-                            <widget class="QPushButton">
+                            <widget class="QPushButton" row="1" column="2">
                                 <property name="name">
                                     <cstring>bankIdButton</cstring>
                                 </property>
@@ -319,6 +323,7 @@
                                     <string>...</string>
                                 </property>
                             </widget>
+			    </grid>
                         </widget>
                         <spacer>
                             <property name="name">
