--- src/frontends/qbanking/lib/Makefile.in     2007-10-15 12:47:51 -0700
+++ src/frontends/qbanking/lib/Makefile.in     2009-08-25 13:27:27 -0700
@@ -556,9 +556,9 @@
 	dialogs/libdialogs.la views/libviews.la \
 	$(aqbanking_internal_libs) $(am__append_1)
 testlib_SOURCES = testlib.cpp
-testlib_LDADD = libqbanking.la $(aqbanking_internal_libs)
+testlib_LDADD = libqbanking.la $(aqbanking_internal_libs) $(qt3_libs)
 checklib_SOURCES = checklib.cpp
-checklib_LDADD = libqbanking.la $(aqbanking_internal_libs)
+checklib_LDADD = libqbanking.la $(aqbanking_internal_libs) $(qt3_libs)
 TESTS = checklib
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
--- src/frontends/qbanking/lib/dialogs/qbimporter.cpp  2009-08-23 15:11:25 -0700
+++ src/frontends/qbanking/lib/dialogs/qbimporter.cpp  2009-08-25 12:53:25 -0700
@@ -835,7 +835,7 @@
     // conversion operator "QCString::operator const char * ()
     // const" that will correctly convert the QCString to the
     // const char*
-    fd=open(fname.local8Bit().data(), O_RDONLY);
+    fd=::open(fname.local8Bit().data(), O_RDONLY);
     if (fd==-1) {
       qs=QWidget::tr("Could not open file \"%1\": %2")
        .arg(fname)
--- Makefile.in        2007-10-15 12:48:21.000000000 -0700
+++ Makefile.in        2009-08-25 14:01:07.000000000 -0700
@@ -548,6 +548,7 @@
 ilocaledir = $(datadir)/locale
 qtlocaledir = $(pkgdatadir)/i18n
 all: $(BUILT_SOURCES) config.h
+	$(MAKE) $(AM_MAKEFLAGS) qt4-port
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
 .SUFFIXES:

