--- a/configure.in	2013-09-30 22:43:48.000000000 -0700
+++ b/configure.in	2013-08-03 09:14:47.000000000 -0700
@@ -3,7 +3,7 @@
 AM_MAINTAINER_MODE
 
 AM_INIT_AUTOMAKE(libdbi-drivers, 0.8.3-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_LIBTOOL_WIN32_DLL
 
 dnl library versioning currently unnecessary...
@@ -63,6 +63,8 @@
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -mv8"
                 CFLAGS="-O20 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8"
                 PROFILE="-pg -g -O20 -D__NO_MATH_INLINES -fsigned-char -mv8" ;;
+	*-*-darwin*)
+		;;
         *)
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
                 CFLAGS="-O20 -D__NO_MATH_INLINES -fsigned-char"

