diff -c /Users/john/gtk/source/libdbi-0.8.3/configure.in\~ /Users/john/gtk/source/libdbi-0.8.3/configure.in
--- a/configure.in	Wed Feb  6 08:03:27 2008
@@ -3,5 +3,5 @@
 AM_MAINTAINER_MODE

 AM_INIT_AUTOMAKE(libdbi, 0.8.3)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)

 dnl Library versioning
 LIB_CURRENT=0
+++ b/configure.in	Mon Sep 30 22:10:00 2013
@@ -61,6 +61,8 @@
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99"
                 CFLAGS="-O20 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99"
                 PROFILE="-pg -g -O20 -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99" ;;
+	*-*-darwin*)
+	    ;;
         *)
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -std=gnu99"
                 CFLAGS="-O20 -D__NO_MATH_INLINES -fsigned-char -std=gnu99"

Diff finished.  Mon Sep 30 22:10:24 2013
