--- Makefile.in~	2009-10-15 14:49:03.000000000 -0700
+++ Makefile.in	2009-10-15 16:39:10.000000000 -0700
--- Makefile.in	2009-01-06 16:30:43.000000000 -0800
+++ Makefile.in	2009-04-19 15:06:17.000000000 -0700
@@ -851,6 +851,8 @@
 info-am:
 
 install-data-am: install-pkgconfigDATA
+	@$(NORMAL_INSTALL)
+	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
 
 install-dvi: install-dvi-recursive
 
@@ -899,7 +899,7 @@
 uninstall-am: uninstall-pkgconfigDATA
 
 .MAKE: $(RECURSIVE_CLEAN_TARGETS) $(RECURSIVE_TARGETS) all \
-	ctags-recursive install-am install-strip tags-recursive
+	ctags-recursive install-am install-data-am install-strip tags-recursive
 
 .PHONY: $(RECURSIVE_CLEAN_TARGETS) $(RECURSIVE_TARGETS) CTAGS GTAGS \
 	all all-am all-local am--refresh check check-am clean \
@@ -911,14 +911,15 @@
 	distclean-hdr distclean-libtool distclean-tags distcleancheck \
 	distdir distuninstallcheck dvi dvi-am html html-am info \
 	info-am install install-am install-data install-data-am \
-	install-dvi install-dvi-am install-exec install-exec-am \
-	install-html install-html-am install-info install-info-am \
-	install-man install-pdf install-pdf-am install-pkgconfigDATA \
-	install-ps install-ps-am install-strip installcheck \
-	installcheck-am installdirs installdirs-am maintainer-clean \
-	maintainer-clean-generic mostlyclean mostlyclean-generic \
-	mostlyclean-libtool pdf pdf-am ps ps-am tags tags-recursive \
-	uninstall uninstall-am uninstall-pkgconfigDATA
+	install-data-hook install-dvi install-dvi-am install-exec \
+	install-exec-am  install-html install-html-am install-info \
+	install-info-am install-man install-pdf install-pdf-am \
+	install-pkgconfigDATA install-ps install-ps-am install-strip \
+	installcheck installcheck-am installdirs installdirs-am \
+	maintainer-clean maintainer-clean-generic mostlyclean \
+	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps \
+	ps-am tags tags-recursive uninstall uninstall-am \
+	uninstall-pkgconfigDATA
 
 
 all-local: Doxyfile
@@ -956,6 +957,10 @@
 @DBUS_GCOV_ENABLED_FALSE@check-coverage:
 @DBUS_GCOV_ENABLED_FALSE@	echo "Need to reconfigure with --enable-gcov"
 
+
+install-data-hook:
+	$(mkinstalldirs) $(DESTDIR)/tmp
+	$(DESTDIR)$(bindir)/dbus-uuidgen --ensure=$(DESTDIR)$(localstatedir)/lib/dbus/machine-id
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
