--- src/bin/gnucash.in  2008-01-07 17:05:50.000000000 -0800
+++ src/bin/gnucash.in  2009-03-22 22:33:05.000000000 -0700
@@ -2,6 +2,11 @@
 
 PATH="@-BIN_DIR-@:${PATH}"
 export PATH
+XDG_CONFIG_DIRS=@-PREFIX-@/etc/xdg
+export XDG_CONFIG_DIRS
+XDG_DATA_DIRS=@-PREFIX-@/share
+export XDG_DATA_DIRS
+
 
 GUILE_WARN_DEPRECATED="no"
 export GUILE_WARN_DEPRECATED
@@ -13,6 +18,7 @@
 GUILE_LOAD_PATH="${EXTRA_PATH}@-PATH_SEPARATOR-@${GUILE_LOAD_PATH}"
 
 EXTRA_LIBS="${GNC_MODULE_PATH}"
+EXTRA_LIBS="${EXTRA_LIBS}:/System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ImageIO.framework/Versions/A/Resources/"
 EXTRA_LIBS="${EXTRA_LIBS}@-PATH_SEPARATOR-@@-GNC_LIB_INSTALLDIR-@"
 EXTRA_LIBS="${EXTRA_LIBS}@-PATH_SEPARATOR-@@-GNC_PKGLIB_INSTALLDIR-@"
 
@@ -24,4 +30,8 @@
 export LD_LIBRARY_PATH
 export DYLD_LIBRARY_PATH
 
+if test -z "$DBUS_SESSION_BUS_ADDRESS"; then
+    eval `dbus-launch --sh-syntax --exit-with-session --config-file=@-PREFIX-@/etc/dbus-1/session.conf`
+fi
+
 exec gnucash-bin "$@"

--- src/bin/Makefile.in 2009-02-23 12:25:48.000000000 -0700
+++ src/bin/Makefile.in 2009-03-22 22:32:32.000000000 -0700
@@ -804,6 +804,7 @@
 gnucash: gnucash.in ${top_builddir}/config.status Makefile
	rm -f $@.tmp
	sed < $< > $@.tmp \
+	    -e 's#@-PREFIX-@#${prefix}#g' \
	    -e 's#@-BIN_DIR-@#${bindir}#g' \
	    -e 's#@-GNC_GUILE_MODULE_DIR-@#${GNC_SHAREDIR}/guile-modules#g' \
	    -e 's#@-GNC_SCM_INSTALL_DIR-@#${GNC_SCM_INSTALL_DIR}#g' \
