--- configure   2009-02-23 12:25:34.000000000 -0800
+++ configure   2009-03-23 11:37:40.000000000 -0700
@@ -24912,7 +24912,7 @@
 # These are unavailable on windows/mingw32
 
 
-for ac_header in X11/Xlib.h glob.h
+for ac_header in glob.h
 do
 as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
--- configure.in        2008-07-06 17:16:42.000000000 -0700
+++ configure.in	2008-07-06 17:17:06.000000000 -0700
@@ -213,7 +213,7 @@
 AM_CONDITIONAL(GNUCASH_SEPARATE_BUILDDIR, test "x${srcdir}" != "x.")
 
 # These are unavailable on windows/mingw32
-AC_CHECK_HEADERS(X11/Xlib.h glob.h)
+AC_CHECK_HEADERS(glob.h)
 AC_CHECK_FUNCS(chown gethostname getppid getuid gettimeofday gmtime_r)
 AC_CHECK_FUNCS(gethostid link)
 
