--- a/libguile/jit.c	2024-02-08 00:05:07
+++ b/libguile/jit.c	2025-09-30 11:16:27
@@ -1414,7 +1414,7 @@
 
       uint8_t *ret = jit_address (j->jit);
 
-#if defined __APPLE__ && HAVE_PTHREAD_JIT_WRITE_PROTECT_NP
+#if defined __APPLE__ && defined __aarch64__ && HAVE_PTHREAD_JIT_WRITE_PROTECT_NP
       pthread_jit_write_protect_np(0);
 #endif
 
@@ -1423,7 +1423,7 @@
       size_t size;
       if (!jit_has_overflow (j->jit) && jit_end (j->jit, &size))
         {
-#if defined __APPLE__ && HAVE_PTHREAD_JIT_WRITE_PROTECT_NP
+#if defined __APPLE__ && defined __aarch64__ && HAVE_PTHREAD_JIT_WRITE_PROTECT_NP
           /* protect previous code arena. leave unprotected after emit()
              since jit_end() also writes to code arena. */
           pthread_jit_write_protect_np(1);
@@ -1442,7 +1442,7 @@
         }
       else
         {
-#if defined __APPLE__ && HAVE_PTHREAD_JIT_WRITE_PROTECT_NP
+#if defined __APPLE__ && defined __aarch64__ && HAVE_PTHREAD_JIT_WRITE_PROTECT_NP
           /* protect previous code arena */
           pthread_jit_write_protect_np(1);
           sys_icache_invalidate(arena->base, arena->size);
