Index: src/os/posix/directory.c
===================================================================
--- src/os/posix/directory.c	(revision 2280)
+++ src/os/posix/directory.c	(working copy)
@@ -28,7 +28,7 @@
 
 #include "directory_p.h"
 #if defined(OS_DARWIN) && defined(ENABLE_LOCAL_INSTALL)
-#  include <CoreFoundation/CFBundle.h>
+#  include <mach-o/dyld.h>
 #  include <sys/param.h>
 #else
 # include "binreloc.h"
@@ -229,9 +229,10 @@
     /* remove "/bin/" or "MacOS" from path */
     s=strrchr(realbuffer, '/');
     if (s) {
-      if (strcasecmp(s, "/bin")==0 ||
-	  strcasecmp(s, "/MacOS")==0)
+      if (strcasecmp(s, "/bin")==0)
 	*s=0;
+      if (strcasecmp(s, "/MacOS")==0)
+	strcpy(s, "/Resources");
     }
   }
 
