*** /usr/local/src/gnome3/libgnomeprint-2.18.6/configure~	Fri Mar  6 02:31:09 2009
--- /usr/local/src/gnome3/libgnomeprint-2.18.6/configure	Sun Mar 22 10:38:33 2009
***************
*** 20084,20090 ****
  esac
  
  
! if test "x$LIBM" = "x"; then
  	{ { $as_echo "$as_me:$LINENO: error: You need -lm(mathematic library) for this version of libgnomeprint." >&5
  $as_echo "$as_me: error: You need -lm(mathematic library) for this version of libgnomeprint." >&2;}
     { (exit 1); exit 1; }; }
--- 20084,20090 ----
  esac
  
  
! if $host != *-*-darwin* && test "x$LIBM" = "x"; then
  	{ { $as_echo "$as_me:$LINENO: error: You need -lm(mathematic library) for this version of libgnomeprint." >&5
  $as_echo "$as_me: error: You need -lm(mathematic library) for this version of libgnomeprint." >&2;}
     { (exit 1); exit 1; }; }
