--- configure.orig      2009-03-06 02:47:56.000000000 -0800
+++ configure   2009-03-21 17:44:49.000000000 -0700
@@ -15866,7 +15866,7 @@
     gnome_keyring_requirement="gnome-keyring-1 >= 0.4"
     GNOME_KEYRING="gnome-keyring-1"
 
-    if test x$GNOME_HAVE_X11 != xyes ; then
+    if test x$have_x != xdisabled && x$GNOME_HAVE_X11 != xyes ; then
         { { $as_echo "$as_me:$LINENO: error: libX11 not found" >&5
 $as_echo "$as_me: error: libX11 not found" >&2;}
    { (exit 1); exit 1; }; }

