--- CMakeLists.txt-org	2010-04-23 23:46:39.000000000 +0200
+++ CMakeLists.txt	2010-04-23 23:47:06.000000000 +0200
@@ -318,7 +318,7 @@
 ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
 INSTALL(DIRECTORY "include/"
-        DESTINATION "include"
+        DESTINATION "include/mysql"
         FILES_MATCHING PATTERN "*.h")
 
 # ----------------------------------------------------------------------
