diff -u /Users/john/Development/Gnucash-Build/Gnucash-3/src/postgresql-9.6.17/configure\~ /Users/john/Development/Gnucash-Build/Gnucash-3/src/postgresql-9.6.17/configure
--- a/configure	2020-02-10 14:21:38.000000000 -0800
+++ b/configure	2020-03-06 17:00:56.000000000 -0800
@@ -13930,6 +13930,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 typedef long long int ac_int64;
 
 /*
@@ -13939,7 +13940,7 @@
 ac_int64 a = 20000001;
 ac_int64 b = 40000005;
 
-int does_int64_work()
+static int does_int64_work()
 {
   ac_int64 c,d;
 
@@ -13953,7 +13954,7 @@
     return 0;
   return 1;
 }
-main() {
+int main() {
   exit(! does_int64_work());
 }
 _ACEOF

Diff finished.  Sat Mar  7 11:17:19 2020
