--- configure.in        2008-07-06 17:16:42.000000000 -0700
+++ configure.in.new    2008-07-06 17:17:06.000000000 -0700
@@ -213,7 +213,7 @@
 AM_CONDITIONAL(GNUCASH_SEPARATE_BUILDDIR, test "x${srcdir}" != "x.")
 
 # These are unavailable on windows/mingw32
-AC_CHECK_HEADERS(X11/Xlib.h glob.h)
+AC_CHECK_HEADERS(glob.h)
 AC_CHECK_FUNCS(chown gethostname getppid getuid gettimeofday gmtime_r)
 AC_CHECK_FUNCS(gethostid link)
 
